src/AtmHttpClient.php

```diff
@@ -0,0 +1,273 @@
+   */
+  public function getPropertySupportedCountries() {
+    $client = new Client();
+
+    $api_key = $this->getHelper()->getApiKey();
+
+    $response = $client->get($this->getBaseUrl() . '/atm-admin/property/supported-countries', [
+      'headers' => [
+        'X-Api-Key' => $api_key,
+      ],
+    ]);
+
+    if ($response->getStatusCode() == 200) {
+      $body = Json::decode(
+          $response->getBody()->getContents()
+      );
+
+      return $body['Countries'];
+    }
+
+    return [];
+  }
+
+  /**
+   * Method provides request to API for generating atm.js.
+   */
+  public function propertyCreate() {
+    $client = new Client();
+
+    $website = $_SERVER['HTTP_HOST'];
+    $name = $this->getHelper()->getApiName();
+    $email = $this->getHelper()->getApiEmail();
+    $country = $this->getHelper()->getApiCountry();
+
+    $price = $this->getHelper()->get('price');
+    $payment_pledged = $this->getHelper()->get('payment_pledged');
+    $currency = $this->getHelper()->get('price_currency');
+    $pledged_type = $this->getHelper()->get('pledged_type');
+
+    $response = $client->put($this->getBaseUrl() . '/atm-admin/property/create', [
+      'headers' => [
+        'X-Api-Key' => $this->getHelper()->getApiKey(),
+      ],
+
+      'json' => [
+        'Name' => $name,
+        'Website' => $website,
+        'SupportEmail' => $email,
+        'Country' => $country,
+        'ConfigDefaults' => [
+          "targetModal" => [
+            "targetCb" => $this->getTargetCbJs(),
+            "toggleCb" => $this->getToggleCbJs(),
+          ],
+          'content' => [
+            'authorCb' => "function(onReady) {onReady({fullName: '$name', avatar: 'https://avatars.io/twitter/mitocgroup'})}",
+            "container" => ".atm--node--view-mode--full",
+            'offset' => $this->getHelper()->get('content_offset'),
+            'lock' => $this->getHelper()->get('content_lock'),
+            'offsetType' => $this->getHelper()->get('content_offset_type'),
+          ],
+          'revenueMethod' => 'micropayments',
+          'ads' => [
+            'relatedVideoCb' => "function (onReady) { }",
+          ],
+          'payment' => [
+            'price' => $price,
+            'pledged' => $payment_pledged,
+            'currency' => $currency,
+            'pledgedType' => $pledged_type,
+          ],
+          'styles' => [
+            'main' => base64_encode(
+              $this->getHelper()->getTemplateOwerallStyles()
+            ),
+          ],
+        ],
+      ],
+    ]);
+
+    if ($response->getStatusCode() == 200) {
+      $_response = Json::decode($response->getBody()->getContents());
+
+      $path_schema = file_default_scheme() . "://atm";
+
+      /** @var \Drupal\Core\File\FileSystem $file_system */
+      $file_system = \Drupal::service('file_system');
+
+      $path = \Drupal::service('file_system')->realpath($path_schema);
+
+      if (!is_dir($path)) {
+        $file_system->mkdir($path, 0777, TRUE);
```

Comment:
Please change permissions to `0644`? (@see http://permissions-calculator.org/decode/0644/)